### PR TITLE
CA-303253: XenMotion breaks VIF connectivity when using DVSC

### PR DESCRIPTION
**Pull request by krizex**
The step of VM rename in migration will break the network connection of the target VM in DVSC scenario. This Pull request aims to fix that.

### CODE REVIEW

File: xc/xenops_server_xen.ml [hunk 1]
```diff
@@ -1199,6 +1199,9 @@ module VM = struct
             begin
               debug "Renaming domain %d from %s to %s" di.Xenctrl.domid vm vm';
               Xenctrl.domain_sethandle xc di.Xenctrl.domid vm';
+              (* TODO: *)
```
Comment (robhoes) [inline xc/xenops_server_xen.ml]: TODO?

File: xc/xenops_server_xen.ml [hunk 2]
```diff
@@ -1199,6 +1199,9 @@ module VM = struct
...
+              let final_uuid_path = "/vm/" ^ vm ^ "/final-uuid" in
+              xs.Xs.rm final_uuid_path;
```
Comment (robhoes) [inline xc/xenops_server_xen.ml]: I think we should make sure that this doesn't fail if the path does not exist. That would make the `rename` function more generally useful.

File: scripts/vif-real [hunk 3]
```diff
@@ -138,14 +138,28 @@ add_to_bridge()
...
     # setup-vif-rules removes the interface from the bridge and clears any rules
     local setup_vif_rules=$(xenstore-read "${PRIVATE}/setup-vif-rules" 2>/dev/null)
+    if [ "${FINAL_DOMUUID}" != "" ];then
+        logger -t scripts-vif "Remove ${dev} of VM:${DOMUUID}->${FINAL_DOMUUID} from bridge"
+        REAL_DOMUUID=${FINAL_DOMUUID}
+    else
+        logger -t scripts-vif "Remove ${dev} of VM:${DOMUUID} from bridge"
+        REAL_DOMUUID=${DOMUUID}
+    fi
     $setup_vif_rules $XENOPSD_BACKEND ${dev} $DOMUUID $DEVID clear \
```
Comment (robhoes) [inline scripts/vif-real]: Use `REAL_DOMUUID` here.

File: xc/xenops_server_xen.ml (hunk 2 above)
Comment (krizex) [reply xc/xenops_server_xen.ml]: However as there is already a `safe_rm` function which we add a wrapper for `Xs.rm`, I would like to use it.

File: scripts/vif-real (hunk 3 above)
Comment (krizex) [reply scripts/vif-real]: Yes.

**Comment by krizex** [comment]
Updated, waiting for BST&BVT results.

**Comment by krizex** [comment]
This change does not work in cross-host migration, more changes need when setting up vif. Updating...

**Comment by krizex** [comment]
Manual and BST/BVT passed.
Please help review again, thanks @robhoes .

**Comment by lindig** [comment]
This looks solid to me.

**Pull request merged**